src/pages/mypage/profile.tsx

```diff
@@ -1,34 +1,101 @@
-import { useState } from 'react';
+import { useMemo, useState } from 'react';
+import { useUserQuery } from '../../api/hook/users';
 import Button from '../../components/common/button';
 import TextInput from '../../components/common/form/text-input';
 import HomeLayout from '../../components/common/layout/layout';
 import { ResignButtonContainer, Section, StyledForm, Title } from '../../styles/mypage/profile';
+import { userIdState } from '../../recoils/atoms/users';
+import { useRecoilValue } from 'recoil';
+import { patchUserInfo, nicknameCheck } from '../../api/fetcher/users';
+import NicknamePopup from '../../components/mypage/mypage-pop-up';
+import { useMutation, useQueryClient } from 'react-query';
+import _, { debounce } from 'lodash';
+
+interface errorMesageType {
+  unknown: string;
+  isError: string;
+  notError: string;
+  [prop: string]: string;
+}
 
 const Profile = () => {
-  const [user, setUser] = useState({
-    id: '1',
-    email: 'user@email.com',
-    nickname: '기존닉네임',
+  const [nickname, setNickname] = useState<string>('');
+  const [error, setError] = useState<string>('');
+  const [nicknamePopupOpen, setNicknamePopupOpen] = useState<boolean>(false);
+
+  const userId = useRecoilValue(userIdState);
+  const queryClient = useQueryClient();
+
+  const { data } = useUserQuery(userId as number, {
+    onSuccess: data => {
+      setNickname(data.nickname);
+    },
+  });
+
+  const errorMessage: errorMesageType = {
+    unknown: '닉네임을 입력해주세요',
+    isError: '이미 등록된 별명입니다',
+    notError: '',
+  };
+
+  Object.freeze(errorMessage);
+
+  const { mutate: patchUserInfoMutate } = useMutation(patchUserInfo, {
+    onSuccess: () => {
+      setError('notError');
+      setNicknamePopupOpen(true);
+      queryClient.invalidateQueries([
+        'USERS',
+        {
+          id: userId,
+        },
```

Comment:
실수로 쿼리키 값을 잘못쓰는 것을 예방하기 위해 useQuery hook을 만들때  `USER_QUER_KEY` 객체를 사용했으므로 
'USERS' 라는 문자열 대신  `USER_QUER_KEY`객체로 사용합니다
```suggestion
      queryClient.invalidateQueries([
        USER_QUER_KEY.USERS,
        {
          id: userId,
        },
```

Reply:
넵 수정하겠습니다!